numpy/lib/__init__.py

```diff
@@ -39,9 +39,9 @@
 from .arrayterator import Arrayterator
 from .arraypad import *
 from ._version import *
-from numpy.core._multiarray_umath import tracemalloc_domain
+from numpy.core._multiarray_umath import tracemalloc_domain, __cpu_features__
```

Comment:
This feels to me like it belongs in core, not lib. Would appreciate the opinion of other maintainers more involved with this.

Reply:
Maybe we could leave it as a private feature for now, so anyone who wants it will have to import it specifically. Then we could just delete these changes

Reply:
alright, CPU features dictionary is no longer exposed, it became private within attr '_CPU_FEATURES'. Is that okay?

Reply:
revert back from this modification and I keep it exposed as `__ cpu_features __` until you make a final decision. to me I'm more than welcome to apply any modification you request.

Reply:
please delete this line, we can decide later where is the best place to expose this publically, if at all

Reply:
It is done, thank you

Reply:
I'd suggest `numpy.__cpu_features__`, would be nice to expose I think.

Reply:
Right, my objection was that the choice was between
* `numpy.lib.__cpu_features__` (this comment chain)
* `numpy.__cpu_features__`
* `numpy.core.__cpu_features__`
and to me only the latter two make sense.

Reply:
`numpy.core` is more or less an accident of history, it really shouldn't be or look public. So I'm basically -1 on adding anything to it.

Reply:
Is `__cpu_features__ ` currently exposed anywhere? I might find this useful to enable the transcendental validation tests added in https://github.com/numpy/numpy/pull/14048

Reply:
It is privately exposed through: `np.core.multiarray._multiarray_umath.__cpu_features__`; Although I would not mind moving it to the main namespace, unless we expect the growth of such functions in some direction which would warrant a new submodule.

Reply:
I just noticed:
```
 'AVX512BITALG': False,
 'AVX512_KNL': False,
```
I assume the spelling difference (underscore vs. no underscore) is not an accident but the way these are spelled? E.g. due to being different feature sets, vs. a specific feature flag? (if that makes sense)

Reply:
@r-devulap, just like @seberg mentioned, its already been used in `test_cpu_features.py`.
@seberg, sure, it's not an accident with underscore it means a group of CPU features,  by `AVX512_KNL` mean the AVX512 CPU features of Intel Knights Landing, the shortcuts are based on Intel compilers and SDE

Reply:
I think underscore exists for keys that represent CPU itself (SKX, KNL, CLX, CNL, ICL etc.) rather than an instruction set (AVX, AVX2, FMA, AVX512F, etc.).